src/plugins/dashboard/public/application/top_nav/editor_menu.tsx

```diff
@@ -0,0 +1,210 @@
+            factories: [factory],
+            panelId: panelCount,
+          };
+
+          panelCount++;
+        }
+      });
+    } else {
+      ungroupedFactories.push(factory);
+    }
+  });
+
+  const getVisTypeMenuItem = (visType: BaseVisType): EuiContextMenuPanelItemDescriptor => {
+    const { name, title, titleInWizard, description, icon = 'empty' } = visType;
+
+    return {
+      name: titleInWizard || title,
+      icon: icon as string,
+      onClick: createNewVisType(visType),
+      'data-test-subj': `visType-${name}`,
+      toolTipContent: description,
+    };
+  };
+
+  const getVisTypeAliasMenuItem = (
+    visTypeAlias: VisTypeAlias
+  ): EuiContextMenuPanelItemDescriptor => {
+    const { name, title, description, icon = 'empty' } = visTypeAlias;
+
+    return {
+      name: title,
+      icon,
+      onClick: createNewVisType(visTypeAlias),
+      'data-test-subj': `visType-${name}`,
+      toolTipContent: description,
+    };
+  };
+
+  const getEmbeddableFactoryMenuItem = (
+    factory: EmbeddableFactoryDefinition
+  ): EuiContextMenuPanelItemDescriptor => ({
+    name: factory.getDisplayName(),
+    icon: 'empty',
+    onClick: async () => {
+      if (trackUiMetric) {
+        trackUiMetric(METRIC_TYPE.CLICK, factory.type);
+      }
+      if (factory.getExplicitInput) {
+        const explicitInput = await factory.getExplicitInput();
+        await dashboardContainer.addNewEmbeddable(factory.type, explicitInput);
+      } else {
+        await factory.create({} as EmbeddableInput, dashboardContainer);
+      }
+    },
+    'data-test-subj': `createNew-${factory.type}`,
+  });
+
+  const editorMenuPanels = [
+    {
+      id: 0,
+      items: [
+        ...visTypeAliases.map(getVisTypeAliasMenuItem),
+        ...Object.values(factoryGroupMap).map(({ id, appName, icon, panelId }) => ({
+          name: appName,
+          icon: icon as EuiContextMenuItemIcon,
+          panel: panelId,
+          'data-test-subj': `dashboardEditor-${id}Group`,
+        })),
+        ...ungroupedFactories.map(getEmbeddableFactoryMenuItem),
+        ...promotedVisTypes.map(getVisTypeMenuItem),
+        {
+          name: i18n.translate('dashboard.editorMenu.aggBasedGroupTitle', {
+            defaultMessage: 'Aggregation based',
+          }),
+          icon: 'visualizeApp',
+          onClick: createNewAggsBasedVis,
+          'data-test-subj': `dashboardEditorAggBasedMenuItem`,
+        },
+      ],
+    },
+    ...Object.values(factoryGroupMap).map(
+      ({ appName, panelId, factories: groupFactories }: FactoryGroup) => ({
+        id: panelId,
+        title: appName,
+        items: groupFactories.map(getEmbeddableFactoryMenuItem),
+      })
+    ),
+  ];
+
+  return (
+    <SolutionToolbarPopover
+      ownFocus
+      label={i18n.translate('dashboard.solutionToolbar.editorMenuButtonLabel', {
+        defaultMessage: 'All editors',
```

Comment:
I think it's inconsistent to label these as "editors", maybe "apps" is a better word? I'm just not aware of any place we use the word "editor"

Reply:
Did you read the design comments above?

Reply:
Yes, I saw that @mdefazio made the same comment about the button label being confusing. It looks like there's been no other discussion of the label here?

Reply:
@ryankeairns had the following suggestions:
> I'm thinking something like 'More visualizations' or 'More chart types'

Reply:
Got it, will this be updated in this PR?

Reply:
Yep! I updated it to "More chart types" in the latest commit